sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -1116,9 +1121,13 @@ cdef class ManhattanDistance{{name_suffix}}(DistanceMetric{{name_suffix}}):
         ITYPE_t size,
     ) except -1 nogil:
         cdef DTYPE_t d = 0
-        cdef cnp.intp_t j
-        for j in range(size):
-            d += fabs(x1[j] - x2[j])
+        cdef cnp.intp_t j, simd_width = 16 / sizeof({{INPUT_DTYPE_t}})
+        cdef cnp.intp_t remainder = size % simd_width
+        if HAS_SIMD:
```

Comment:
Naive question: is there a benefit to making this a "compile time if statement"? And if yes, is that even possible/advisable?

Reply:
After reading (and thinking) a bit more, in particular the Numpy setup, I think you have to have a runtime check. One that is even more "runtime" than this one which relies on a compile time constant. (Maybe except for SSE SSE2 as this seems to be the "min" setting for numpy?)